README.md

```diff
@@ -51,16 +51,112 @@ See [nvim-dap](https://github.com/mfussenegger/nvim-dap/wiki/Debug-Adapter-insta
+| otherwise                                   | any                 | same as ID                            |
+
+You can overwrite this behaviour by providing a custom mapping function during setup:
+
+```lua
+require("neotest").setup({
+  adapters = {
+    require("neotest-rust") {
+        parameterized_test_discovery = "cargo"
+        resolve_case_name = function(id, macro, file)
+            -- id:    string the test case identifier returned by `cargo nextest list`
+            -- macro: string the (first) macro name which makes this test parameterized (e.g. `values`, `files`, `test_case`, ...)
+            -- file:  path the path to the file under test
+            local name = ...
+            return name
+        end,
+    }
+  }
+})
+
+```
+
+
+
 ## Limitations
 
 The following limitations apply to both running and debugging tests.
 
 - Assumes unit tests in `main.rs`, `mod.rs`, and `lib.rs` are in a `tests`
   module.
-- Does not support `rstest`'s `#[case]` macro.
 - When running tests for a `main.rs` in an integration test subdirectory (e.g.
   `tests/testsuite/main.rs`), all tests in that subdirectory will be run (e.g.
   all tests in `tests/testsuite/`). This is because Cargo lacks the capability
   to specify a test file.
+- When using `#[test_case(...)]` for parameterized tests and `treesitter` as
+  discovery mode, make sure to _always_ use a test comment (e.g.
+  `#[test_case(arg1, arg2, ... ; "test_comment")]`). Otherwise the test runner
+  might not find this case. With `cargo` as strategy this is not strictly
+  necessary.
+- When using for example `#[rstest::files("**/*.txt")]` in a test the discovery
+  tries to heuristically "guess" each file and render it in the summary view.
+  This only works, though, if the file exists relative to the project root
+  and contains no special characters except underscores (and `.` before the file
+  extension). Otherwise the name resolution is just skipped and the `cargo`
+  id is displayed. Has no influence on the running of each test case, only for
+  the name in the summary. ![_](./media/rstest-files-naming.png)
+- When using a combination of `rstest` features the _first_ one in the argument
+  list always decides on the heuristic name resolution. For example if you have
+  a test like `fn foo(#[case] x: i32, #[files("*")] file: PathBuf) {}` the name
+  resolution will try to resolve the test ID based on the `case` attribute, which
+  is not consistent with the `files` rendering. Try swapping arguments if necessary.
+  Has no influence on the running of each test case, only for the name in the summary
+- When using `rstest` try to avoid additional attributes like `#[tokio::test]`
+  otherwise the test discovery does not work correctly. You shouldn't need these
+  attributes anyway, since [rstest supports async tests out of the box](https://docs.rs/rstest/latest/rstest/attr.rstest.html#async)
```

Comment:
this isn't right is it? my read of the docs is that async-std is supported out of the box, but for everything else you still need to add the attribute

Reply:
True, with the new treesitter query, this shouldn't be a constraint anymore. Now you can place the `#[tokio::test]` where you want